app/services/customers/eu_auto_taxes_service.rb

```diff
@@ -4,14 +4,18 @@
 
 module Customers
   class EuAutoTaxesService < BaseService
-    def initialize(customer:)
+    def initialize(customer:, new_record:, changed_attributes:)
```

Comment:
I automatically assumed `changed_attributes` was a list of attributes and maybe even their before/after values.

I'd rename `changed_attributes` to something like `tax_attributes_changed` (something ending with _changed).

I like `new_record` because it matches the active model `new_record?` method.

Reply:
changed attribute name 👍